### PR TITLE
feat: renaming kv directory to metadata

### PR DESCRIPTION
**Pull request by Lilit0x**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?
The KV directory has been renamed to metadata to provide more useful info for users.

## Checklist

- [x]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

#2493

### CODE REVIEW

File: src/common/config/src/lib.rs [hunk 1]
```diff
@@ -46,7 +46,7 @@ impl Default for WalConfig {
 }
 
 pub fn kv_store_dir(store_dir: &str) -> String {
```
Comment (killme2008) [inline src/common/config/src/lib.rs:48]: Let's rename the function too.

Comment (Lilit0x) [reply src/common/config/src/lib.rs:48]: Okay

**Review by killme2008 — APPROVED** [review]
LGTM

**Comment by killme2008** [comment]
@waynexia Please take a look.

**Review by waynexia — APPROVED** [review]
LGTM, thanks @Lilit0x

**Pull request merged**